### PR TITLE
Add name change in affiliationHistory

### LINKED ISSUE #99: Update name in AffiliationHistory after Name edits
... for discussion

### PR DESCRIPTION
**Pull request by shuenj**

### CODE REVIEW

File: src/test/java/seedu/address/model/affiliation/AffiliationModifierTest.java [hunk 1]
```diff
@@ -78,4 +78,23 @@ public void nameChangeAffiliation_validParam_success() {
                 patientB.getName(), newName, model);
         assertTrue(doctorA.getAffiliations().contains(new Affiliation(newName.fullName)));
     }
+
+    @Test
+    public void nameChangeAffiliationHistory_validParam_success() {
+        Person doctorAlice = new DoctorBuilder().withName("Alice")
+            .withAffiliationHistory("Bob").build();
```
Comment (leezhanpeng) [inline src/test/java/seedu/address/model/affiliation/AffiliationModifierTest.java]: Maybe to increase test coverage, you can add another `AffiliationHistory` here, say "Thomas", for Alice? i.e. someone that is not in the model.

File: src/test/java/seedu/address/model/affiliation/AffiliationModifierTest.java [hunk 2]
```diff
@@ -78,4 +78,23 @@ public void nameChangeAffiliation_validParam_success() {
...
+        Person doctorCharlie = new DoctorBuilder().withName("Charlie")
+            .withAffiliationHistory("Bob").build();
+        Person patientBob = new PatientBuilder().withName("Bob")
+            .withAffiliations("Alice", "Charlie")
+            .withAffiliationHistory("Alice", "Charlie").build();
+        model.addPerson(doctorAlice);
+        model.addPerson(doctorCharlie);
+        model.addPerson(patientBob);
```
Comment (leezhanpeng) [inline src/test/java/seedu/address/model/affiliation/AffiliationModifierTest.java:96]: Would it be possible to add another `Doctor` into the model, where the doctor is in the `AffiliationHistory` of Bob, but itself has an empty `AffiliationHistory`? I understand this is an impossible scenario, but we will be able to hit the test coverage for line #L132 in `EditCommand` with this.

**Review by leezhanpeng — COMMENTED** [review]
LGTM! Suggested some changes so that the test coverage can further increase to pass the checks. Maybe we can see if those suggestions are useful before we approve this PR.

File: src/test/java/seedu/address/model/affiliation/AffiliationModifierTest.java (hunk 2 above)
Comment (shuenj) [reply src/test/java/seedu/address/model/affiliation/AffiliationModifierTest.java:96]: Sure, will do!

File: src/test/java/seedu/address/model/affiliation/AffiliationModifierTest.java (hunk 1 above)
Comment (shuenj) [reply src/test/java/seedu/address/model/affiliation/AffiliationModifierTest.java]: Added, thaanks!

**Review by leezhanpeng — APPROVED** [review]
LGTM!!

**Pull request merged**